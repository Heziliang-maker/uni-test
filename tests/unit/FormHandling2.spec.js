/*
 * @Date: 2021-11-16
 * @Description:
 */

import { mount } from "@vue/test-utils";

import FormHandling2 from "@/components/FormHandling2";

// 使用各种表单元素
// test("submits a form", async () => {
//   const wrapper = mount(FormHandling2);

//   await wrapper.find("input[type=email]").setValue("my@email.com");

//   await wrapper.find("textarea").setValue("Lorem ipsum dolor sit amet");

//   await wrapper.find("select").setValue("new-york");

//   await wrapper.find("input[type=checkbox]").setChecked(true);

//   await wrapper.find("input[type=radio][value=weekly]").setChecked();

//   expect(wrapper.vm.$el).toMatchSnapshot();
// });

// 触发复杂的事件监听器
test("submits the form", async () => {
  const wrapper = mount(FormHandling2);

  await wrapper.find("input[type=email]").setValue("my@email.com");

  await wrapper.find("textarea").setValue("Lorem ipsum dolor sit amet");

  await wrapper.find("select").setValue("new-york");

  await wrapper.find("input[type=checkbox]").setChecked(true);

  await wrapper.find("input[type=radio][value=weekly]").setChecked();

  await wrapper.find("form").trigger("submit.prevent");

  expect(wrapper.emitted()).toHaveProperty("submit");

  expect(wrapper.emitted("submit")[0][0]).toStrictEqual({
    email: "my@email.com",
    description: "Lorem ipsum dolor sit amet",
    city: "new-york",
    subscribe: true,
    interval: "weekly",
  });
});
