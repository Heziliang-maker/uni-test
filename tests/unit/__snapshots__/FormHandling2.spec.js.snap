// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submits a form 1`] = `
<form>
  <input
    type="email"
  />
   
  <textarea />
   
  <select>
    <option
      value="new-york"
    >
      New York
    </option>
     
    <option
      value="moscow"
    >
      Moscow
    </option>
  </select>
   
  <input
    type="checkbox"
  />
   
  <input
    type="radio"
    value="weekly"
  />
   
  <input
    type="radio"
    value="monthly"
  />
   
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
